static-pod-check/static-pod-check-workaround.sh

```diff
@@ -0,0 +1,288 @@
+#!/bin/bash
+#
+# Periodically checks static pod container states.
+# If unhealthy for a period of time, restart kubelet.service
+#
+
+PROG=$(basename "$0")
+
+# Tag for logs in journald
+declare logtag="static-pod-check-workaround"
+
+# Debug logs
+declare verbose="no"
+
+# Log to stdout instead of journald
+declare log_to_stdout="no"
+
+# Location of static pod manifests
+declare manifests_dir="/etc/kubernetes/manifests"
+
+# Minimum time kubelet should be active before we check states
+declare kubelet_up_minimum_secs=1200
+
+# Time between checks when static pods are healthy
+declare interval_healthy_secs=300
+
+# Time between checks when static pods are unhealthy
+declare interval_unhealthy_secs=30
+
+# Number of failed health checks for a given pod before restarting kubelet
+declare max_consecutive_failures=20
+
+# For testing, force health check failure by creating this flag file
+declare test_failure_flag_file=/tmp/force_static_pod_check_failure.flag
+
+# For testing, allow dry-run to only log
+declare dryrun="no"
+
+# Associative array for tracking consecutive failures
+declare -A health_failures
+
+#
+# Logging functions
+#
+function loginfo {
+    if [ "${log_to_stdout}" = "no" ]; then
+        logger -t "${logtag}" --id=$BASHPID "$1"
+    else
+        echo "$(date): $1"
+    fi
+}
+
+function logdebug {
+    if [ "${verbose}" = "no" ]; then
+        return
+    fi
+
+    if [ "${log_to_stdout}" = "no" ]; then
+        logger -t "${logtag}" --id=$BASHPID "$1"
+    else
+        echo "$(date): $1"
+    fi
+}
+
+#
+# Verify kubelet has been running for at least kubelet_up_minimum_secs seconds
+#
+function kubelet_running {
+    if ! systemctl -q is-active kubelet.service ; then
+        logdebug "kubelet.service is not active"
+        return 1
+    fi
+
+    local kubelet_start_time
+    local kubelet_start_time_secs
+    local secs_since_start
+
+    kubelet_start_time=$(systemctl show kubelet.service --no-page | awk -F'=' '$1 == "ActiveEnterTimestamp" { print $2}')
```

Comment:
This is cleaner:
systemctl show kubelet.service --property ActiveEnterTimestamp --value

Reply:
Looks good